### DISCUSSION TITLE
Is there an equivalent to SharePoint "ensure User" in the SDK?

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/pnpcore#317] [Q&A] [answered]
I am planning to user this SDK but I can't find a "EnsureUser" method anywhere in the documentation.
Is there another way to get the same results.

the specific use case is to add a person to a list person field having just his email address.

Thank you much

**Comment by koltyakov** [comment, marked as answer]
Yep, there is [EnsureUser](https://github.com/pnp/pnpcore/blob/ed1f0694818db71d788ef7e443bd2d97155f957d/src/sdk/PnP.Core/Model/SharePoint/Core/Internal/Web.cs#L544).

**Comment by Ofer-Gal** [comment]
Thanks
